apps/sms/views/shared/js/settings.js

```diff
@@ -72,6 +79,15 @@ var Settings = {
     }
   },
 
+  // return resolve promise with sendReadReport status
+  initSendReadReport: function initSendReadReport(value) {
+    this._sendReadReportDefer.resolve(value);
+    delete this._sendReadReportDefer;
+
+    navigator.mozSettings.addObserver('messages.mms.sendReadReport.enabled',
+      (e) => this.sendReadReport = Promise.resolve(e.settingValue));
```

Comment:
I think it's a good thing to keep everything related to settings in one place.

So that means we could still have a function in Settings that requests mozSettings and returns a promise.

But I also think it's good to cache the value because currently we may call the function "markMessagesRead" several time (if we call markThreadRead for several threads).
Maybe a fix for this could be to make markThreadRead accept a list of threads so that markMessagesRead is called only once ?

In the future we'll use a function that returns a promise instead of a property that holds a promise anyway so we'll have to do this work... But I feel it's outside the scope of this bug.

From my point of view, the code is nearly ready here so I'd prefer to not change much more, except for the text change and maybe reverting the meaning (but reverting it in Settings only).